modules/rentals_data_manipulation.rb

```diff
@@ -0,0 +1,46 @@
+require 'json'
+
+module RentalsDataManipulation
+  def save_rentals_to_json(file_path, new_rental)
+    parsed_data = []
+
+    if File.exist?(file_path)
+      current_data = File.read(file_path)
+
+      parsed_data = JSON.parse(current_data) unless current_data.empty?
+    end
```

Comment:
- [ ] Hey there! 👋 Great job on implementing the rental-saving method. 📚 I noticed you're checking if the rentals.json file exists before reading from it, which is awesome. However, one thing to consider for better robustness: it might be a good idea to refactor the code to create the db directory and the rentals.json file if they don't exist.

This adjustment can prevent potential errors and ensure a smoother user experience. 🚀 Keep up the good work! If you have any questions or need further clarification, feel free to ask. 👍